pandas/io/orc.py

```diff
@@ -50,7 +50,9 @@ def read_orc(
 
     import pyarrow.orc
 
-    path, _, _, _ = get_filepath_or_buffer(path)
-    orc_file = pyarrow.orc.ORCFile(path)
+    ioargs = get_filepath_or_buffer(path)
+    orc_file = pyarrow.orc.ORCFile(ioargs.filepath_or_buffer)
     result = orc_file.read(columns=columns, **kwargs).to_pandas()
+    if ioargs.should_close:
+        ioargs.filepath_or_buffer.close()
```

Comment:
I added that but I think we never reach that because [pyarrow.orc.ORCFile](https://arrow.apache.org/docs/python/generated/pyarrow.orc.ORCFile.html) can only handle strings and some pyarrow object. If that is true, the type annotation for `path` is wrong and I will remove the close call!